pyvista/utilities/fileio.py

```diff
@@ -122,12 +125,18 @@ def standard_reader_routine(reader, filename, attrs=None):
         value.
 
     """
+    observer = pyvista.utilities.errors.Observer()
+    observer.observe(reader)
+
     if attrs is None:
         attrs = {}
     if not isinstance(attrs, dict):
         raise TypeError('Attributes must be a dictionary of name and arguments.')
     if filename is not None:
-        reader.SetFileName(filename)
+        if hasattr(reader, 'SetCaseFileName'):
+            reader.SetCaseFileName(filename)
+        else:
+            reader.SetFileName(filename)
```

Comment:
Can we change this to a try/except with an `AttributeError`?

Reply:
Done!